ui/lib/kv/addon/components/page/secret/details.hbs

```diff
@@ -2,15 +2,63 @@
   <:tabLinks>
     <LinkTo @route="secret.details" data-test-secrets-tab="Secret">Secret</LinkTo>
     <LinkTo @route="secret.metadata.index" data-test-secrets-tab="Metadata">Metadata</LinkTo>
-    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Metadata">Version History</LinkTo>
-    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Metadata">Version Diff</LinkTo>
+    <LinkTo @route="secret.metadata.versions" data-test-secrets-tab="Version History">Version History</LinkTo>
+    <LinkTo @route="secret.metadata.diff" data-test-secrets-tab="Version Diff">Version Diff</LinkTo>
   </:tabLinks>
 
   <:toolbarFilters>
-    {{! TODO add JSON toggle }}
+    <Toggle
+      data-test-json-view-toggle
+      @name="json"
+      @status="success"
+      @size="small"
+      @checked={{this.showJsonView}}
+      @onChange={{this.toggleJsonView}}
+    >
+      <span class="has-text-grey">JSON</span>
+    </Toggle>
   </:toolbarFilters>
-
   <:toolbarActions>
     <ToolbarLink @route="secret.edit" @type="add">Create new version</ToolbarLink>
   </:toolbarActions>
-</KvPageHeader>
\ No newline at end of file
+</KvPageHeader>
+
+{{#if this.showJsonView}}
+  <div class="has-top-margin-s">
+    <JsonEditor @title="Version data" @value={{stringify @model.secret.secretData}} @readOnly={{true}} />
+  </div>
+{{else}}
+  <div class="info-table-row-header">
+    <div class="info-table-row thead">
+      <div class="th column is-one-quarter">
+        Key
+      </div>
+      <div class="th column">
+        Value
+      </div>
+      <div class="th column justify-right" data-test-created-time>
+        {{#if @model.secret.createdTime}}
```

Comment:
Would it be possible to also show the version? I know this is not the current functionality but it was something we talked about adding as a nice-to-have because it's now available on the data endpoint.